src/collectors/loggly/loggly.py

```diff
@@ -0,0 +1,268 @@
+                    (metric, self.configfile))
+
+            set_defaults(config)
+
+            config.update({
+                'search_URI': 'https://{0.subdomain}.loggly.com'
+                              '/apiv2/search?q={0.query}&from='
+                              '-{1.interval}s&until=now'.format(
+                                  config, self.config_bunch),
+                'events_URI': 'https://{0.subdomain}.loggly.com'
+                              '/apiv2/events?rsid=%s'.format(config)})
+
+    def collect(self):
+        """Collect metrics from loggly."""
+        def parse_timestamp(metric_config, timestamp):
+            if metric_config.timestamp_format is 'int':
+                return datetime.datetime.fromtimestamp(
+                    timestamp / 10**(ceil(log10(timestamp)) - 10))
+            else:
+                return datetime.datetime.strptime(timestamp,
+                                                  self.config_bunch
+                                                  .timestamp_format)
+
+        def retrieve_results(metric_config):
+            def now_as_timefield():
+                if metric_config.timestamp_format is 'int':
+                    return time.mktime(datetime.datetime.now().timetuple())
+                else:
+                    return datetime.datetime.now().strftime(
+                        metric_config.timestamp_format)
+
+            def log_error_and_continue(code):
+                known_errors = {
+                    400: self.log.error('Loggly returned error 400:'
+                                        'Bad Request'),
+                    401: self.log.error('Loggly returned error 401:'
+                                        'Unauthorized'),
+                    403: self.log.error('Loggly returned error 403:'
+                                        'Forbidden'),
+                    404: self.log.error('Loggly returned error 404:'
+                                        'Not Found'),
+                    409: self.log.error('Loggly returned error 409:'
+                                        'Duplicate'),
+                    410: self.log.error('Loggly returned error 410:'
+                                        'Gone'),
+                    500: self.log.error('Loggly returned error 500:'
+                                        'Internal Server Error'),
+                    501: self.log.error('Loggly returned error 501:'
+                                        'Not Implemented'),
+                    503: self.log.error('Loggly returned error 503:'
+                                        'Throttled'),
+                    504: self.log.error('Loggly returned error 504:'
+                                        'Gateway Timeout')
+                }
+                try:
+                    known_errors[code]
+                except KeyError:
+                    self.log.error('Loggly returned unrecognized error code %s'
```

Comment:
Same as above, this comes from Loggly's API documentation, there are the values they have special meanings for.